### PR TITLE
EN6-86 content settings metadata

### PR DESCRIPTION
**Pull request by jeffgo10**
contains metadata in content settings

### CODE REVIEW

File: src/state/content-settings/__tests__/actions.test.js [hunk 1]
```diff
@@ -185,4 +235,103 @@ describe('contentSettings thunks', () => {
       done();
     }).catch(done.fail);
   });
+
+  it('sendPostMetadataMap', (done) => {
+    store.dispatch(sendPostMetadataMap('key', 1)).then(() => {
+      expect(postMetadataMap).toHaveBeenCalledWith({ key: 'key', mapping: 1 });
+      const actions = store.getActions();
```
Comment (raxkaynan) [inline src/state/content-settings/__tests__/actions.test.js]: It seems to lack length assertion.
```suggestion
      const actions = store.getActions();
      expect(actions).toHaveLength(3);
```

File: src/state/content-settings/__tests__/actions.test.js [hunk 2]
```diff
@@ -185,4 +235,103 @@ describe('contentSettings thunks', () => {
...
+      expect(actions[2].payload.id).toEqual('postMetadataMap');
+      done();
+    }).catch(done.fail);
+  });
+
+  it('sendPostMetadataMap error', (done) => {
+    postMetadataMap.mockImplementationOnce(mockApi({ errors: true }));
+    store.dispatch(sendPostMetadataMap('key', 2)).then(() => {
+      expect(postMetadataMap).toHaveBeenCalledWith({ key: 'key', mapping: 2 });
+      const actions = store.getActions();
```
Comment (raxkaynan) [inline src/state/content-settings/__tests__/actions.test.js]: Same with [this](https://github.com/entando/entando-cms/pull/24/files#r334602809).
```suggestion
      const actions = store.getActions();
      expect(actions).toHaveLength(3);
```

File: src/state/content-settings/__tests__/actions.test.js [hunk 3]
```diff
@@ -185,4 +235,103 @@ describe('contentSettings thunks', () => {
...
+      done();
+    }).catch(done.fail);
+  });
+
+  it('sendPutMetadataMap', (done) => {
+    const key = 'key';
+    const mapping = 1;
+    store.dispatch(sendPutMetadataMap(key, mapping)).then(() => {
+      expect(putMetadataMap).toHaveBeenCalledWith(key, mapping);
+      const actions = store.getActions();
```
Comment (raxkaynan) [inline src/state/content-settings/__tests__/actions.test.js]: Same with [this](https://github.com/entando/entando-cms/pull/24/files#r334602809).
```suggestion
      const actions = store.getActions();
      expect(actions).toHaveLength(3);
```

File: src/state/content-settings/__tests__/actions.test.js [hunk 4]
```diff
@@ -185,4 +235,103 @@ describe('contentSettings thunks', () => {
...
+    }).catch(done.fail);
+  });
+
+  it('sendPutMetadataMap error', (done) => {
+    const key = 'key';
+    const mapping = 2;
+    putMetadataMap.mockImplementationOnce(mockApi({ errors: true }));
+    store.dispatch(sendPutMetadataMap(key, mapping)).then(() => {
+      expect(putMetadataMap).toHaveBeenCalledWith(key, mapping);
+      const actions = store.getActions();
```
Comment (raxkaynan) [inline src/state/content-settings/__tests__/actions.test.js]: Same with [this](https://github.com/entando/entando-cms/pull/24/files#r334602809).
```suggestion
      const actions = store.getActions();
      expect(actions).toHaveLength(1);
```

File: src/state/content-settings/__tests__/actions.test.js [hunk 5]
```diff
@@ -185,4 +235,103 @@ describe('contentSettings thunks', () => {
...
+      expect(actions[0]).toHaveProperty('type', 'errors/add-errors');
+      done();
+    }).catch(done.fail);
+  });
+
+  it('checkAndPutMetadataMap', (done) => {
+    const values = { legend: 'awt, two' };
+    store.dispatch(checkAndPutMetadataMap(values)).then(() => {
+      expect(putMetadataMap).toHaveBeenCalledWith('legend', 'awt, two');
+      const actions = store.getActions();
```
Comment (raxkaynan) [inline src/state/content-settings/__tests__/actions.test.js]: Same with [this](https://github.com/entando/entando-cms/pull/24/files#r334602809).
```suggestion
      const actions = store.getActions();
      expect(actions).toHaveLength(3);
```

File: src/state/content-settings/__tests__/actions.test.js [hunk 6]
```diff
@@ -185,4 +235,103 @@ describe('contentSettings thunks', () => {
...
+      done();
+    }).catch(done.fail);
+  });
+
+  it('checkAndPutMetadataMap error', (done) => {
+    const values = { legend: 'awt, three' };
+    putMetadataMap.mockImplementationOnce(mockApi({ errors: true }));
+    store.dispatch(checkAndPutMetadataMap(values)).then(() => {
+      expect(putMetadataMap).toHaveBeenCalledWith('legend', 'awt, three');
+      const actions = store.getActions();
```
Comment (raxkaynan) [inline src/state/content-settings/__tests__/actions.test.js]: Same with [this](https://github.com/entando/entando-cms/pull/24/files#r334602809).
```suggestion
      const actions = store.getActions();
      expect(actions).toHaveLength(3);
```

File: src/state/content-settings/__tests__/actions.test.js [hunk 7]
```diff
@@ -185,4 +235,103 @@ describe('contentSettings thunks', () => {
...
+      expect(actions[1]).toHaveProperty('type', 'errors/add-errors');
+      expect(actions[2]).toHaveProperty('type', TOGGLE_LOADING);
+      done();
+    }).catch(done.fail);
+  });
+
+  it('sendDeleteMetadataMap', (done) => {
+    store.dispatch(sendDeleteMetadataMap('key')).then(() => {
+      expect(deleteMetadataMap).toHaveBeenCalledWith('key');
+      const actions = store.getActions();
```
Comment (raxkaynan) [inline src/state/content-settings/__tests__/actions.test.js]: Same with [this](https://github.com/entando/entando-cms/pull/24/files#r334602809).
```suggestion
      const actions = store.getActions();
      expect(actions).toHaveLength(3);
```

File: src/state/content-settings/__tests__/actions.test.js [hunk 8]
```diff
@@ -185,4 +235,103 @@ describe('contentSettings thunks', () => {
...
+      expect(actions[2].payload.id).toEqual('deleteMetadataMap');
+      done();
+    }).catch(done.fail);
+  });
+
+  it('sendDeleteMetadataMap error', (done) => {
+    deleteMetadataMap.mockImplementationOnce(mockApi({ errors: true }));
+    store.dispatch(sendDeleteMetadataMap('key')).then(() => {
+      expect(deleteMetadataMap).toHaveBeenCalledWith('key');
+      const actions = store.getActions();
```
Comment (raxkaynan) [inline src/state/content-settings/__tests__/actions.test.js]: Same with [this](https://github.com/entando/entando-cms/pull/24/files#r334602809).
```suggestion
      const actions = store.getActions();
      expect(actions).toHaveLength(3);
```

**Comment by jeffgo10** [comment]
hi @raxkaynan, applied your suggestions. thanks!

**Pull request merged**